src/datamodel/turn/turn_history.rs

```diff
@@ -0,0 +1,5 @@
+use crate::datamodel::turn::finished_turn::FinishedTurn;
+
+pub struct TurnHistory<'a> {
```

Comment:
this kind of syntax thrills me. haha, but it looks correct.

Reply:
To have a lifetime in the struct? This is required if the struct holds a reference, because the reference needs to have a lifetime. This will probably be replaced by smart pointers `Rc<T>` instead of references, but I propose to do that refactor in a separate PR: make a pass through the full app and design, and replace references by smart pointers.

Reply:
Yeah It'll probably change to a Rc or Box, no doubt.

Reply:
OK, but do you agree to revisit the smart pointers in a later PR? I think it is OK as it is for this one.

Reply:
Yes, I agree with that :-)